pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/GetOptionsForActivityInstanceQuestionRoute.java

```diff
@@ -0,0 +1,121 @@
+        String participantGuid = request.params(RouteConstants.PathParam.USER_GUID);
+        String instanceGuid = request.params(RouteConstants.PathParam.INSTANCE_GUID);
+        String questionStableId = request.params(RouteConstants.PathParam.QUESTION_STABLE_ID);
+
+        DDPAuth ddpAuth = RouteUtil.getDDPAuth(request);
+        String operatorGuid = StringUtils.defaultIfBlank(ddpAuth.getOperator(), participantGuid);
+        boolean isStudyAdmin = ddpAuth.hasAdminAccessToStudy(studyGuid);
+        String autoCompleteQuery = request.queryParams(RouteConstants.QueryParam.TYPEAHEAD_QUERY);
+        String queryLimit = request.queryParams(RouteConstants.QueryParam.TYPEAHEAD_QUERY_LIMIT);
+        int limit = (StringUtils.isNotBlank(queryLimit) && Integer.valueOf(queryLimit) <= DEFAULT_LIMIT)
+                ? Integer.valueOf(queryLimit) : DEFAULT_LIMIT;
+        ContentStyle style = RouteUtil.parseContentStyleHeaderOrHalt(request, response, ContentStyle.STANDARD);
+        LanguageDto preferredUserLanguage = RouteUtil.getUserLanguage(request);
+
+        LOG.info("Fetching auto complete picklist options for activity instance {} and participant {} in study {} by operator {} "
+                + "(isStudyAdmin={})", instanceGuid, participantGuid, studyGuid, operatorGuid, isStudyAdmin);
+
+        RemoteAutoCompleteResponse result = TransactionWrapper.withTxn(handle -> {
+
+            //Get all options first from ActivityDef (already in memory) then render and match
+            long timestamp = Instant.now().toEpochMilli();
+            long langCodeId = preferredUserLanguage.getId();
+            List<PicklistOption> suggestions = new ArrayList<>();
+            List<PicklistOption> allOptions = getPicklistOptions(handle, studyGuid, instanceGuid, questionStableId);
+            if (StringUtils.isBlank(autoCompleteQuery)) {
+                LOG.info("Option suggestion query is blank, returning all results size to default limit");
+                suggestions = allOptions.stream().limit(limit).collect(Collectors.toList());
+                renderer.bulkRenderAndApply(handle, allOptions, style, langCodeId, timestamp);
+            } else {
+                renderer.bulkRenderAndApply(handle, allOptions, style, langCodeId, timestamp);
+                suggestions = filterOptions(allOptions, autoCompleteQuery, limit);
+            }
+
+            return new RemoteAutoCompleteResponse(autoCompleteQuery, suggestions);
+        });
+
+        return result;
+    }
+
+    private List<PicklistOption> getPicklistOptions(Handle handle, String studyGuid,
+                                                    String activityInstanceGuid, String questionStableId) {
+
+        ActivityInstanceDto instanceDto = handle.attach(JdbiActivityInstance.class)
+                .getByActivityInstanceGuid(activityInstanceGuid)
+                .orElseThrow(() -> new DaoException("Could not find activity instance using guid " + activityInstanceGuid
```

Comment:
Would this result in a 500 error code? let's save those for truly unexepected server errors. This is probably closer to a 422.